scripts/install

```diff
@@ -1,202 +1,234 @@
 
-# Install dependencies using Helpers
-ynh_package_install_from_equivs ../conf/z-push-deps.control \
-|| ynh_die "Unable to install dependencies"
+ynh_app_setting_set $app domain $domain
+ynh_app_setting_set $app path $path
 
+ynh_app_setting_set $app statedir "$statedir"
+ynh_app_setting_set $app final_logpath "$final_logpath"
 
-final_path="/var/www/$app"
+#=================================================
+# INSTALL DEPENDENCIES
+#=================================================
+ynh_print_info "Installing dependencies..."
 
-CHECK_FINALPATH () {    # Vérifie que le dossier de destination n'est pas déjà utilisé.
-        if [ -e "$final_path" ]
-        then
-                echo "This path already contains a folder" >&2
-                false
-        fi
-}
+ynh_install_app_dependencies $pkg_dependencies
 
-CHECK_FINALPATH
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+ynh_print_info "Setting up source files..."
 
-statedir="/home/yunohost.app/$app"
+ynh_app_setting_set $app final_path $final_path
+# Download, check integrity, uncompress and patch the source from app.src
+mkdir "$final_path"
+mkdir "$final_path/tmp"
+ynh_setup_source "$final_path/tmp"
+cp -a "$final_path/tmp/src/." "$final_path/."
+rm -r "$final_path/tmp"
 
-CHECK_STATEDIRPATH () {    # Vérifie que le dossier de destination n'est pas déjà utilisé.
-        if [ -e "$statedir" ]
-        then
-                echo "This path already contains a folder" >&2
-                false
-        fi
-}
+#=================================================
+# NGINX CONFIGURATION
+#=================================================
+ynh_print_info "Configuring nginx web server..."
 
-CHECK_STATEDIRPATH
+### `ynh_add_nginx_config` will use the file conf/nginx.conf
 
+# Create a dedicated nginx config
+ynh_add_nginx_config
 
-# Define variables and Save app settings
-ynh_app_setting_set "$app" domain "$domain"
-ynh_app_setting_set "$app" final_path "$final_path"
-ynh_app_setting_set "$app" statedir "$statedir"
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+ynh_print_info "Configuring php-fpm..."
 
-final_logpath="/var/log/$app"
-ynh_app_setting_set "$app" final_logpath "$final_logpath"
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
 
-final_nginxconf="/etc/nginx/conf.d/${domain}.d/${app}.conf"
-ynh_app_setting_set "$app" final_nginxconf "$final_nginxconf"
+#=================================================
+# SPECIFIC SETUP
+#=================================================
+# CREATE STATEDIR AND FINAL_LOGPATH
+#=================================================
+ynh_print_info "Creating statedir and final_logpath..."
 
-final_phpconf="/etc/php5/fpm/pool.d/${app}.conf"
-ynh_app_setting_set "$app" final_phpconf "$final_phpconf"
+mkdir -p $statedir
+chown -R www-data:www-data $statedir
```

Comment:
As you're using php fpm, first you should use a dedicated user if possible.
And here give permission to this user instead of www-data.
So you will restrict the permission of this user to this directory only in case of compromising.

Reply:
Switched to ad dedicated user